lib/tasks/multitenant/tenants.rake

```diff
@@ -68,6 +68,37 @@ namespace :multitenant do
       update_tenant_ids(proc { |object| object.account.tenant_id }, proc { account }, proc { tenant_id == nil }, args.to_hash.merge({ table_name: 'Alert' }))
     end
 
+    desc 'Check and remove orphaned objects (whose tenant is missing), pass "destroy" argument to delete'
+    task :cleanup_orphans, [:mode] => :environment do |_task, args|
+      puts 'Checking and removing orphaned objects...'
+
+      destroy = args[:mode] == "destroy"
+
+      puts "WARNING: the found orphan objects will be destroyed" if destroy
+
+      provider_account_ids = Account.where(provider: true).pluck(:id) + [Account.master.id]
+
+      ApplicationRecord.descendants.each do |model|
+        next unless model.table_exists? && model.column_names.include?('tenant_id')
+
+        orphaned_objects = model.where.not(tenant_id: provider_account_ids)
+
+        if orphaned_objects.exists?
+          puts "Found orphaned objects in #{model.table_name}:"
+          orphaned_objects.find_each { |obj| puts "- ID: #{obj.id}, Tenant ID: #{obj.tenant_id}" }
+
+          if destroy
+            puts "Destroying orphan #{model.table_name}..."
+            orphaned_objects.in_batches(of: 100).destroy_all
+          end
+        else
+          puts "No orphaned objects found in #{model.table_name}."
```

Comment:
Maybe better this:

```suggestion
          puts "No orphaned objects found in #{model.name}."
```

Because many models share the same table:

```
...
No orphaned objects found in fields_definitions.
No orphaned objects found in cms_sections.
No orphaned objects found in cms_sections.
No orphaned objects found in cms_templates.
No orphaned objects found in cms_templates.
No orphaned objects found in cms_templates.
No orphaned objects found in cms_templates.
No orphaned objects found in cms_templates.
No orphaned objects found in cms_templates.
No orphaned objects found in cms_templates.
No orphaned objects found in cms_templates.
No orphaned objects found in cms_templates.
No orphaned objects found in cms_templates.
No orphaned objects found in cms_templates.
No orphaned objects found in cms_templates.
No orphaned objects found in cms_templates.
No orphaned objects found in users.
No orphaned objects found in cinstances.
No orphaned objects found in cinstances.
No orphaned objects found in cinstances.
No orphaned objects found in cinstances.
No orphaned objects found in referrer_filters.
...
```

This way we can know which model is actually being cleaned up:
```
...
No orphaned objects found in FieldsDefinition.
No orphaned objects found in CMS::Section.
No orphaned objects found in CMS::Builtin::Section.
No orphaned objects found in CMS::Template.
No orphaned objects found in CMS::Layout.
No orphaned objects found in CMS::BasePage.
No orphaned objects found in CMS::Builtin.
No orphaned objects found in CMS::Builtin::StaticPage.
No orphaned objects found in CMS::Builtin::Page.
No orphaned objects found in CMS::Page.
No orphaned objects found in CMS::Partial.
No orphaned objects found in CMS::Builtin::Partial.
No orphaned objects found in CMS::Builtin::LegalTerm.
No orphaned objects found in CMS::Portlet.
No orphaned objects found in CMS::Portlet::Base.
No orphaned objects found in CMS::EmailTemplate.
No orphaned objects found in User.
No orphaned objects found in Contract.
No orphaned objects found in Cinstance.
No orphaned objects found in ServiceContract.
No orphaned objects found in AccountContract.
No orphaned objects found in ReferrerFilter.

...
```

Same for line 87:

```
puts "Found orphaned objects in #{model.name}:"
```